src/main/resources/testchipip/csrc/uart.cc

```diff
@@ -0,0 +1,138 @@
+//See LICENSE for license details
+
+#include "uart.h"
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#define _XOPEN_SOURCE
+#include <stdlib.h>
+#include <stdio.h>
+
+#ifndef _WIN32
+#include <unistd.h>
+
+// name length limit for ptys
+#define PTYNAMELEN 256
+
+/* There is no "backpressure" to the user input for sigs. only one at a time
+ * non-zero value represents unconsumed special char input.
+ *
+ * Reset to zero once consumed.
+ */
+
+// This is fine for multiple UARTs because UARTs > uart 0 will use pty, not stdio
+char specialchar = 0;
+
+void sighand(int s) {
+    switch (s) {
+        case SIGINT:
```

Comment:
Aren't people going to be annoyed when other signals are ignored?
Or is there some reason we can ignore other signals (like ctrl-D)?

Reply:
Again, the purpose of this widget is mainly to read from the UART and not interact with it. With that being said, this is currently what FireSim does so I copied that. @sagark might know a particular reason why this is ignoring most of the other signals.